### ISSUE TITLE
False negative when referencing outer-scoped symbol in a method if the method is using PEP695 type parameter syntax

### ISSUE THREAD

**Issue by luwqz1** [microsoft/pyright#9280]
```python
class Foo:
    def foo(self, bar: Bar) -> None: ...  # Error: "Bar" is not defined

    def bar[T](self, bar: Bar) -> None: ...  # No error, but a runtime error will occur


class Bar: ...
```

[pyright playground](https://pyright-play.net/?pythonVersion=3.13&strict=true&code=MYGwhgzhAEBiD28BcBYAUNT0AmBTAZtPogBQS4j4A00ARmAE5LQBCjAlNALQB80AcvAB2uZgDoJ6dFhwE6jANoAVALpkK1eU1YdufQSPGS0UtKEgw22iWPRA)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.386
